benchmarks/timeseries/benchmark_datasourcing.py

```diff
@@ -0,0 +1,170 @@
+async def benchmark_data_sourcing(
+    num_evc_chargers: int, num_msgs_per_battery: int
+) -> None:
+    """Benchmark the data sourcing actor.
+
+    Args:
+        num_evc_chargers: number of EV Chargers to create for the mock microgrid.
+        num_msgs_per_battery: number of messages to send out for each battery.
+
+    Returns:
+        A tuple of (number of messages sent, time taken to stream the messages).
+    """
+    num_expected_messages = (
+        num_evc_chargers * len(COMPONENT_METRIC_IDS) * num_msgs_per_battery
+    )
+    mock_grid = MockMicrogrid(
+        grid_side_meter=False, num_values=num_msgs_per_battery, sample_rate_s=0.0
+    )
+
+    mock_grid.add_ev_chargers(num_evc_chargers)
+    mock_grid.start_mock_client(enable_mock_client)
+
+    request_channel = Broadcast[ComponentMetricRequest](
+        "DataSourcingActor Request Channel"
+    )
+
+    channel_registry = ChannelRegistry(name="Microgrid Channel Registry")
+    request_receiver = request_channel.new_receiver(
+        "datasourcing-benchmark", maxsize=(num_evc_chargers * len(COMPONENT_METRIC_IDS))
+    )
+    request_sender = request_channel.new_sender()
+
+    consume_tasks = []
+
+    start_time = perf_counter()
+    samples_sent = 0
+
+    for evc_id in mock_grid.evc_ids:
+        for component_metric_id in COMPONENT_METRIC_IDS:
+            request = ComponentMetricRequest(
+                "current_phase_requests", evc_id, component_metric_id, None
+            )
+
+            recv_channel = channel_registry.new_receiver(request.get_channel_name())
+
+            async def consume(channel: Receiver[Any]) -> None:
+                while True:
+                    try:
+                        await channel.ready()
+                        channel.consume()
+                    except ReceiverStoppedError:
+                        return
+
+                    nonlocal samples_sent
+                    samples_sent += 1
+
+            await request_sender.send(request)
+            consume_tasks.append(asyncio.create_task(consume(recv_channel)))
+
+    DataSourcingActor(request_receiver, channel_registry)
+
+    await asyncio.gather(*consume_tasks)
+
+    time_taken = perf_counter() - start_time
+
+    await mock_grid.cleanup()
+
+    print(f"Samples Sent: {samples_sent}, time taken: {time_taken}")
+    print(f"Samples per second: {samples_sent / time_taken}")
+    print(
+        "Expected samples: "
+        f"{num_expected_messages}, missing: {num_expected_messages - samples_sent}"
+    )
+    print(
+        f"Missing per EVC: {(num_expected_messages - samples_sent) / num_evc_chargers}"
+    )
+
+
+def parse_args() -> Tuple[int, int, bool]:
+    """Parse the command line arguments.
+
+    Returns:
+        A tuple of (num evc chargers, num messages per battery, record allocations).
+    """
+    parser = argparse.ArgumentParser(description="Benchmark the data sourcing actor.")
+    parser.add_argument(
+        "num_evc_chargers",
```

Comment:
why `num_evc_chargers` and not `num_ev_chargers`? I personally found a bit redundant the `c` in `evc` but I might be missing something?

Reply:
If this comment applies then the replacement should be done for each definition of the variable